### PR TITLE
[Docs] Add migration.md and notice for default branch switching

### PR DESCRIPTION
**Pull request by JingweiZhang12**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: README_zh-CN.md [hunk 1]
```diff
@@ -47,6 +47,8 @@
 
 **新闻**：
 
+**默认分支已经从 `master` 切换到 `1.x`。我们鼓励用户迁移到最新版本，请参考 [迁移指南](https://mmdetection3d.readthedocs.io/zh_CN/dev-1.x/migration.html) 以了解更多细节。**
+
```
Comment (ZCMax) [inline README_zh-CN.md:51]: The type of link for migration guide doc is not unified in English and Chinese doc

File: docs/en/migration.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Along with the release of OpenMMLab 2.0, MMDetection3D (namely MMDet3D) 1.x made many significant changes, resulting in less redundant, more efficient code and a more consistent overall design. These changes break backward compatibility. Therefore, we prepared this migration guide to make the transition as smooth as possible so that all users can enjoy the productivity benefits of the new MMDet3D and the entire OpenMMLab 2.0 ecosystem.
```
Comment (ZCMax) [inline docs/en/migration.md:1]: 1.x -> 1.1

File: docs/en/migration.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Environment
+
+MMDet3D 1.1 depends on the new foundational library for training deep learning models [MMEngine](https://github.com/open-mmlab/mmengine), and therefore has an entirely different dependency chain compared with MMDet3D 1.0. Even if you have a well-rounded MMDet3D 1.0 environment before, you still need to create a new python environment for MMDet3D 1.1. We provide a detailed [installation guide](./get_started.md) for reference.
```
Comment (ZCMax) [inline docs/en/migration.md:5]: Det3D 1.1 depends on the new foundational library [MMEngine](https://github.com/open-mmlab/mmengine) for training deep learning models,

Comment (ZCMax) [reply docs/en/migration.md:5]: compared with MMDet3D 1.0 and 0.x

Comment (ZCMax) [inline docs/en/migration.md:5]: Even if you have a well-rounded MMDet3D 1.0 / 0.x environment before

File: docs/en/migration.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+## Dataset
+
+You should update the annotation files generated in the old version since some key words and structures of annotation in MMDet3D 1.x have changed. The updating script is the following (taking KITTI as an xample):
```
Comment (ZCMax) [inline docs/en/migration.md:9]: Whether should we offer two guidance for users in 0.x and 1.0 respectively?

File: docs/en/migration.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+```python
+python tools/dataset_converters/update_infos_to_v2.py
+        --dataset kitti
+        --pkl-path ./data/kitti/kitti_infos_train.pkl
+        --out-dir ./kitti_v2/
+```
+
+## Model
+
+MMDet3D 1.x supports loading weights trained by the old version. For models that are important or highly used, we totally verified the precision of them in the 1.x version. Espectially for some models that has potential performance drop or training bugs in the old version, such as [centerpoint](https://github.com/open-mmlab/mmdetection3d/issues/2390), we check them and make sure the right precision in the new version. If you find any problem, please feel free to raise an [Issue](https://github.com/open-mmlab/mmdetection3d/issues). On the other hand, we also add some the latest SOTA motheds in our [package](../../configs/) and [projects](../../projects/). Thus, we strongly recommend you can use MMDet3D 1.1 to implement your project.
```
Comment (ZCMax) [inline docs/en/migration.md:20]: 1.x -> 1.1

Comment (ZCMax) [inline docs/en/migration.md:20]: MMDet3D 1.1 now supports loading weights trained by the old version (1.0 version). For models that are important or frequently used, we have thoroughly verified their precision in the 1.1 version. Especially for certain models like [[centerpoint](https://github.com/open-mmlab/mmdetection3d/issues/2390)](https://github.com/open-mmlab/mmdetection3d/issues/2390)https://github.com/open-mmlab/mmdetection3d/issues/2390) that may experience potential performance drops or training bugs in the old version, we have checked and ensured their accuracy in the new version. If you encounter any problems, please feel free to raise an [[Issue](https://github.com/open-mmlab/mmdetection3d/issues)](https://github.com/open-mmlab/mmdetection3d/issues). Additionally, we have added some of the latest SOTA methods into our [[package](https://github.com/open-mmlab/mmdetection3d/configs/)](../../configs/) and [[projects](https://github.com/open-mmlab/mmdetection3d/projects/)](../../projects/), making MMDet3D 1.1 a highly recommended choice for implementing your project.

File: docs/en/migration.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Along with the release of OpenMMLab 2.0, MMDetection3D (namely MMDet3D) 1.1 made many significant changes, resulting in less redundant, more efficient code and a more consistent overall design. These changes break backward compatibility. Therefore, we prepared this migration guide to make the transition as smooth as possible so that all users can enjoy the productivity benefits of the new MMDet3D and the entire OpenMMLab 2.0 ecosystem.
+
+## Environment
+
+MMDet3D 1.1 depends on the new foundational library for training deep learning models [MMEngine](https://github.com/open-mmlab/mmengine), and therefore has an entirely different dependency chain compared with MMDet3D 1.0. Even if you have a well-rounded MMDet3D 1.0 / 0.x environment before, you still need to create a new python environment for MMDet3D 1.1. We provide a detailed [installation guide](./get_started.md) for reference.
```
Comment (ZCMax) [inline docs/en/migration.md:5]: MMDet3D 1.1 depends on the new foundational library [MMEngine](https://github.com/open-mmlab/mmengine)  for training deep learning models

File: docs/en/migration.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+You should update the annotation files generated in the 1.0 version since some key words and structures of annotation in MMDet3D 1.1 have changed. The updating script is the following (taking KITTI as an xample):
+
+```python
+python tools/dataset_converters/update_infos_to_v2.py
+        --dataset kitti
+        --pkl-path ./data/kitti/kitti_infos_train.pkl
+        --out-dir ./kitti_v2/
+```
+
+If your annotation files are generated in the 0.x version, you should firstly update them to 1.0 version using this [script](../../tools/update_data_coords.py). Alternatively, you can re-genetate annotation files from scratch use this [script](../../tools/create_data.py).
```
Comment (ZCMax) [inline docs/en/migration.md:18]: use -> using

File: docs/en/migration.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        --dataset kitti
+        --pkl-path ./data/kitti/kitti_infos_train.pkl
+        --out-dir ./kitti_v2/
+```
+
+If your annotation files are generated in the 0.x version, you should firstly update them to 1.0 version using this [script](../../tools/update_data_coords.py). Alternatively, you can re-genetate annotation files from scratch using this [script](../../tools/create_data.py).
+
+## Model
+
+MMDet3D 1.1 supports loading weights trained by the old version (1.0 version). For models that are important or frequently used, we have thoroughly verified the precision of them in the 1.1 version. Espectially for some models that may experience potential performance drop or training bugs in the old version, such as [centerpoint](https://github.com/open-mmlab/mmdetection3d/issues/2390), we have checked them and ensured the right precision in the new version. If you encounter any problem, please feel free to raise an [Issue](https://github.com/open-mmlab/mmdetection3d/issues). Additionally, we have added some of the latest SOTA motheds in our [package](../../configs/) and [projects](../../projects/), making MMDet3D 1.1 a highly recommended choice for implementing your project.
```
Comment (Tai-Wang) [inline docs/en/migration.md:22]: Especially

**Review by Tai-Wang — CHANGES_REQUESTED** [review]
Please use Grammarly (at least as a plug-in) to check and polish the writing

**Review by ZCMax — APPROVED** [review]

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**